apollo-compiler/src/test/graphql/com/example/input_object_type/TestQueryExpected.java

```diff
@@ -0,0 +1,86 @@
+package com.example.input_object_type;
+
+import com.apollostack.api.Query;
+import java.lang.Override;
+import java.lang.String;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+public final class TestQuery implements Query<TestQuery.Variables> {
```

Comment:
On iOS, `ApolloClient` defines `fetch(query:)`, which takes a `GraphQLQuery`, and `perform(mutation:)`, which takes  a `GraphQLMutation`. There is no difference in implementation right now, but the idea is that we do need to differentiate between operation types because they support different options. Mutations shouldn't go through the cache for example, and it doesn't make sense to `watch` mutations.